specification/evaluation-context/evaluation-context.md

```diff
@@ -1,5 +1,41 @@
 # Evaluation Context
 
+**Status**: [Experimental](../README.md#document-statuses)
+
 ## Overview
 
-// TODO: this is a stub.
+The `evaluation context` provides ambient information for the purposes of flag evaluation. Contextual data may be used as the basis for targeting, including rule-based evaluation, overrides for specific subjects, or fractional flag evaluation.
+
+### Fields
+
+##### Requirement 3.1
+
+> The `evaluation context` structure **MUST** define a required `targeting key` field of type string, identifying the subject of the flag evaluation.
+
+The targeting key uniquely identifies the subject (end-user, or client service) of a flag evaluation. Providers may use this as a unique identifier for fractional-flags, rules, or overrides targeting specific users.
+
+##### Requirement 3.2
+
+> The evaluation context **MUST** define the following optional fields: `email` (string), `first name` (string), `last name`(string), `name`(string), `client ip`(string), `tz`(string), `locale`(string), `date`(string | date), `environment`(string), `application`(string), and `version`(string).
+
+See the table below for details.
+
+| Name        | Type           | Purpose                                            | Example(s)                                                |
+| ----------- | -------------- | -------------------------------------------------- | --------------------------------------------------------- |
+| email       | string         | Subject RFC 5322 conformant email                  | "first.last@example.com"                                  |
+| first name  | string         | Subject given name                                 | "Jane"                                                    |
+| last name   | string         | Subject last name                                  | "Smith"                                                   |
+| name        | string         | Subject full name and titles and suffixes          | "Ms. Jane Smith"                                          |
+| client ip   | string         | Client IP address (IPv4 or IPv6)                   | "216.7.132.218","2001:0db8:85a3:0000:0000:8a2e:0370:7334" |
```

Comment:
Is `remote address` a better name? Unleash uses that, Launchdarkly uses `ip`. `ip` seems vague, I'd like to imply that we mean a client IP, not the host in the case of server-side flag evaluation.

Reply:
previous c constructs used to use just `addr`. I think the 'remote' is implied, so maybe just `address`?

Reply:
`address` could be confused with a home address. Especially, with the information collected above such as name and email address. I actually don't mind `ip`  in this situation.

Reply:
ip is fine

Reply:
I will change this to `ip`